File: 6-zigzag-conversion/zigzag-conversion.java
class Solution {
    public String convert(String s, int numRows) {
        if(numRows == 1) return s;

        String[] rows = new String[numRows];
        for(int i = 0 ; i < numRows ; i++){
            rows[i] = "";
        }

        int row = 0 ;
        boolean flag = true;

        for(char c : s.toCharArray()){
            rows[row] += c;

            if(row == 0) flag = true;
            if(row == numRows - 1) flag = false;
            row += flag ? 1 : -1;
        }
        String result = "";
        for(String r : rows){
            result += r;
        }
        return result;

    }
}

File: 6-zigzag-conversion/zigzag-conversion.cs
public class Solution {
    public string Convert(string s, int numRows) {
        if(numRows == 1){
            return s;
        }
        var sb = new StringBuilder[numRows];
        var c = 0 ; 
        var d = 1;
        for(int i = 0 ; i < sb.Length ; i++){
            sb[i] = new StringBuilder();
        }
        for(int i = 0 ; i < s.Length ; i++){
            sb[c].Append(s[i]);
            if(c == numRows - 1){
                d = -1;
            }
            else if(c == 0){
                d = 1;
            }
            c =c + d;
            
        }
        StringBuilder result = new StringBuilder();
        for(int i = 0 ; i < numRows ; i++){
            result.Append(sb[i]);
        }
        return result.ToString();
    }
}